python/ray/data/_internal/execution/interfaces/physical_operator.py

```diff
@@ -139,9 +145,18 @@ def on_data_ready(self, max_bytes_to_read: Optional[int]) -> int:
                 break
 
             try:
+                if meta_ref.is_nil():
+                    meta_ref = self._streaming_gen._next_sync(1)
+                if meta_ref.is_nil():
+                    self._pending_block_pair = (block_ref, meta_ref)
+                    break
                 meta_with_schema: "BlockMetadataWithSchema" = ray.get(
-                    next(self._streaming_gen)
+                    meta_ref, timeout=1
                 )
+            except ray.exceptions.GetTimeoutError:
+                logger.warning(f"Get Meta timeout for (block_ref={block_ref.hex()})")
```

Comment:
Could you make this warning more descriptive? I don't think regular users would know how to interpret this

Maybe something like:
> Getting block metadata timed out for (block_ref={block_ref.hex()}). This can happen if a node dies after a task produces a block but before it has produced metadata.